### PR TITLE
Encode targets for Time Series Classification Pipelines

### LINKED ISSUE #1976: Fix TS Pipelines to encode target data for both predict and predict_proba
Minimum repro:
```python
from evalml.pipelines import TimeSeriesBinaryClassificationPipeline
import pandas as pd

class TSBinaryPipeline(TimeSeriesBinaryClassificationPipeline):
        """Logistic Regression Pipeline for time series binary classification problems."""
        component_graph = ['Delayed Feature Transformer', 'Logistic Regression Classifier']

X = pd.DataFrame({"A": [i for i in range(1000)]})
y = pd.Series(["String target {}".format(i % 2) for i in range(1000)])
ts_binary_pipeline = TSBinaryPipeline(parameters={"Logistic Regression Classifier": {"n_jobs": 1}, "pipeline": {"gap": 0, "max_delay": 0}})
ts_binary_pipeline.fit(X, y)
pred_proba = ts_binary_pipeline.predict_proba(X, y)
```
Results in:
image

This issue tracks fixing this (likely adding the call to the target encoder in `predict` and `predict_proba`), then adding in an equivalent test as `test_binary_pipeline_string_target_thresholding` in `test_pipelines` to the `time_series_pipeline` tests.

### PR DESCRIPTION
**Pull request by freddyaboulton**
### Pull Request Description
Fixes #1976

-----
*After creating the pull request: in order to pass the **release_notes_updated** check you will need to update the "Future Release" section of* `docs/source/release_notes.rst` *to include this pull request by adding :pr:`123`.*

### CODE REVIEW

**Review by bchen1116 — APPROVED** [review]
Nice! Can you add a test for thresholding the binary TS pipeline? As per issue, "then adding in an equivalent test as `test_binary_pipeline_string_target_thresholding` in `test_pipelines` to the `time_series_pipeline` tests."

Otherwise, looks good to me!

**Review by ParthivNaresh — APPROVED** [review]
Looks great! Nice mocking work

**Comment by freddyaboulton** [comment]
@bchen1116 Great call! Just pushed up that test.

File: evalml/tests/pipeline_tests/test_pipelines.py [hunk 1]
```diff
@@ -1976,14 +1976,19 @@ def test_score_error_when_custom_objective_not_instantiated(problem_type, logist
...
     y = make_data_type('ww', pd.Series([f"String value {i}" for i in y]))
     objective = get_objective("F1", return_instance=True)
-    pipeline = logistic_regression_binary_pipeline_class(parameters={"Logistic Regression Classifier": {"n_jobs": 1}})
-    pipeline.fit(X, y)
-    assert pipeline.threshold is None
-    pred_proba = pipeline.predict_proba(X, y).iloc[:, 1]
-    pipeline.optimize_threshold(X, y, pred_proba, objective)
-    assert pipeline.threshold is not None
+    for pipeline_class in [time_series_binary_classification_pipeline_class,
+                           logistic_regression_binary_pipeline_class]:
```
Comment (chukarsten) [inline evalml/tests/pipeline_tests/test_pipelines.py:1987]: Might be a nit, but it'll look better in test failure, perhaps paratetrize this one?

**Review by chukarsten — APPROVED** [review]
Awesome, love the tests too.  I dunno what it is about a good test that I can look at the name of and read and see it do what it set out to do that makes me feel good, but it does.  I added a note about maybe parameterizing your for loop to make test failures a little prettier, but take it or leave it.

File: evalml/tests/pipeline_tests/test_pipelines.py (hunk 1 above)
Comment (freddyaboulton) [reply evalml/tests/pipeline_tests/test_pipelines.py]: Good call! I just got lazy cause you can't parametrize over fixtures lol

**Pull request merged**